packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -821,6 +823,105 @@ export const getWorkflow = async (id: string, projectId?: string) => {
 	return response?.data ?? null;
 };
 
+// FIXME: These follow functions overlap with WorkflowWrapper, will need to be sorted out. Jan 2025
+export const newOperator = (
+	workflowId: string,
+	op: Operation,
+	pos: Position,
+	options: { size?: OperatorNodeSize; state?: any }
+) => {
+	let currentUserName: string | undefined = '';
+	try {
+		currentUserName = useAuthStore().user?.username;
+	} catch (err) {
+		// do nothing
+	}
+	const nodeSize: Size = getOperatorNodeSize(options.size ?? OperatorNodeSize.medium);
+
+	const operator: WorkflowNode<any> = {
+		id: uuidv4(),
+		workflowId,
+		operationType: op.name,
+		displayName: op.displayName,
+		documentationUrl: op.documentationUrl,
+		imageUrl: op.imageUrl,
+		x: pos.x,
+		y: pos.y,
+
+		createdBy: currentUserName,
+		createdAt: Date.now(),
+
+		active: null,
+		state: options.state ?? {},
+		uniqueInputs: op.uniqueInputs ?? false,
+
+		inputs: op.inputs.map((port) => ({
+			id: uuidv4(),
+			type: port.type,
+			label: port.label,
+			status: WorkflowPortStatus.NOT_CONNECTED,
+			value: null,
+			isOptional: port.isOptional ?? false
+		})),
+
+		outputs: op.outputs.map((port) => ({
+			id: uuidv4(),
+			type: port.type,
+			label: port.label,
+			status: WorkflowPortStatus.NOT_CONNECTED,
+			value: null,
+			isOptional: false,
+			state: {}
+		})),
+
+		status: OperatorStatus.INVALID,
+		width: nodeSize.width,
+		height: nodeSize.height
+	};
+	if (op.initState && _.isEmpty(operator.state)) {
+		operator.state = op.initState();
+	}
+	return operator;
+};
+
+export const selectOutput = async (id: string, nodeId: string, outputId: string, projectId?: string) => {
+	console.log('>> workflowService.selectOutput');
+	const response = await API.post(`/workflows/${id}/select-output/${nodeId}/${outputId}`, {
+		params: { 'project-id': projectId }
+	});
+	return response?.data ?? null;
+};
+
+export const appendOutput = async (id: string, nodeId: string, output: WorkflowOutput<any>, nodeState: any) => {
+	console.log('>> workflowService.appendOutput');
+	const response = await API.post(`/workflows/${id}/append-output/${nodeId}`, { output, nodeState });
+	return response?.data ?? null;
+};
+
+export const addNode = async (id: string, node: WorkflowNode<any>) => {
+	console.log('>> workflowService.addNode');
+	const response = await API.post(`/workflows/${id}/node`, node);
+	return response?.data ?? null;
+};
+
+export const addEdge = async (id: string, edge: WorkflowEdge) => {
+	console.log('>> workflowService.addEdge', edge.source, edge.target);
+	const response = await API.post(`/workflows/${id}/edge`, edge);
+	return response?.data ?? null;
+};
+
+export const removeNodes = async (id: string, nodes: string[]) => {
```

Comment:
I think it'd be more clear to call these nodeIds
```suggestion
export const removeNodes = async (id: string, nodeIds: string[]) => {
```

Reply:
done